app/src/main/java/com/example/perfectweatherallyear/ui/detailWeather/DetailWeatherFragment.kt

```diff
@@ -6,34 +6,56 @@ import android.view.View
 import android.view.ViewGroup
 import androidx.fragment.app.Fragment
 import androidx.fragment.app.viewModels
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.example.perfectweatherallyear.appComponent
 import com.example.perfectweatherallyear.databinding.FragmentDetailWeatherBinding
 import com.example.perfectweatherallyear.model.DayWeather
+import com.example.perfectweatherallyear.util.NotificationUtil
 import com.google.gson.GsonBuilder
+import javax.inject.Inject
 
 const val ARG_DAY_WEATHER: String = "DAY_WEATHER"
-
 class DetailWeatherFragment : Fragment() {
-    private val detailWeatherViewModel by viewModels<DetailWeatherViewModel>()
+    private lateinit var detailWeatherForecastAdapter: DetailWeatherForecastAdapter
+    private lateinit var binding: FragmentDetailWeatherBinding
+    lateinit var dayWeather: DayWeather
+
+    @Inject
+    lateinit var detailWeatherViewModelFactory: DetailWeatherViewModelFactory
+
+    private val detailWeatherViewModel by viewModels<DetailWeatherViewModel>{
+        detailWeatherViewModelFactory
+    }
 
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View {
-        val binding = FragmentDetailWeatherBinding.inflate(inflater, container, false)
-        binding.lifecycleOwner = viewLifecycleOwner
-        binding.detailViewModel = detailWeatherViewModel
+        binding = FragmentDetailWeatherBinding.inflate(layoutInflater)
+        requireContext().appComponent.inject(this)
+
+        val builder = GsonBuilder()
+        val gson = builder.create()
+        dayWeather = gson.fromJson(arguments?.getString(ARG_DAY_WEATHER), DayWeather::class.java)
```

Comment:
Maybe passing only some ID of dayWeather is enought?

Reply:
I need dayWeather.id, dayWeather.cityId and dayWeather.date to get hourlyWeather